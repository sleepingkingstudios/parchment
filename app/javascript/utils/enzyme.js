import { valueOrDefault } from './object';
import { safeCapitalize } from './string';

export const extractInjectedComponent = (component) => {
  const injected = component.shallow().find('ComponentWithInjectedProps');

  return injected.exists() ? injected : component;
};

export const toHaveFormControl = expect => (received, expectedControl, withProps) => {
  const formControl = received.find(expectedControl);
  const expectedProps = valueOrDefault(withProps, {});

  expect(formControl).toExist();
  expect(formControl).toHaveProp(expectedProps);

  return { pass: true, message: () => '' };
};

export const toHaveFormGroup = expect => (received, expectedGroup, fn) => {
  const formGroup = received.find(expectedGroup);

  expect(formGroup).toExist();

  fn(extractInjectedComponent(formGroup).shallow());

  return { pass: true, message: () => '' };
};

export const toHaveFormInput = expect => (received, props) => {
  const {
    form,
    inputType,
    prop,
  } = props;

  expect(received).toHaveProp({ prop });

  const formInput = received.find(`${safeCapitalize(prop)}Input`);

  expect(formInput).toExist();
  expect(formInput).toHaveProp({ form });

  const shallowInput = formInput.shallow();

  expect(shallowInput).toExist();
  expect(shallowInput).toHaveDisplayName(valueOrDefault(inputType, 'FormInput'));

  return { pass: true, message: () => '' };
};
