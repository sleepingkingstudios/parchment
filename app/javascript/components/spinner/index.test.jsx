import React from 'react';
import { shallow } from 'enzyme';

import Spinner from './index';

describe('<Spinner />', () => {
  const defaultProps = {};

  it('should render an inline spinner', () => {
    const rendered = shallow(<Spinner {...defaultProps} />);

    expect(rendered).toHaveDisplayName('span');
    expect(rendered).toHaveClassName('spinner-border');
    expect(rendered).toHaveProp('role', 'status');
    expect(rendered).toHaveProp('aria-hidden', 'true');
  });

  it('should match the snapshot', () => {
    const rendered = shallow(<Spinner {...defaultProps} />);

    expect(rendered).toMatchSnapshot();
  });

  describe('with className: value', () => {
    const props = { ...defaultProps, className: 'text-danger' };

    it('should render an inline spinner', () => {
      const rendered = shallow(<Spinner {...props} />);
      const { className } = props;

      expect(rendered).toHaveDisplayName('span');
      expect(rendered).toHaveClassName('spinner-border');
      expect(rendered).toHaveClassName(className);
    });
  });

  describe('with size: lg', () => {
    const props = { ...defaultProps, size: 'lg' };

    it('should render an inline spinner', () => {
      const rendered = shallow(<Spinner {...props} />);

      expect(rendered).toHaveDisplayName('span');
      expect(rendered).toHaveClassName('spinner-border');
      expect(rendered).toHaveClassName('spinner-border-lg');
    });
  });

  describe('with size: sm', () => {
    const props = { ...defaultProps, size: 'sm' };

    it('should render an inline spinner', () => {
      const rendered = shallow(<Spinner {...props} />);

      expect(rendered).toHaveDisplayName('span');
      expect(rendered).toHaveClassName('spinner-border');
      expect(rendered).toHaveClassName('spinner-border-sm');
    });
  });

  describe('with spinnerStyle: grow', () => {
    const props = { ...defaultProps, spinnerStyle: 'grow' };

    it('should render an inline spinner', () => {
      const rendered = shallow(<Spinner {...props} />);

      expect(rendered).toHaveDisplayName('span');
      expect(rendered).toHaveClassName('spinner-grow');
    });

    describe('with size: lg', () => {
      const propsWithSize = { ...props, size: 'lg' };

      it('should render an inline spinner', () => {
        const rendered = shallow(<Spinner {...propsWithSize} />);

        expect(rendered).toHaveDisplayName('span');
        expect(rendered).toHaveClassName('spinner-grow');
        expect(rendered).toHaveClassName('spinner-grow-lg');
      });
    });

    describe('with size: sm', () => {
      const propsWithSize = { ...props, size: 'sm' };

      it('should render an inline spinner', () => {
        const rendered = shallow(<Spinner {...propsWithSize} />);

        expect(rendered).toHaveDisplayName('span');
        expect(rendered).toHaveClassName('spinner-grow');
        expect(rendered).toHaveClassName('spinner-grow-sm');
      });
    });
  });
});
