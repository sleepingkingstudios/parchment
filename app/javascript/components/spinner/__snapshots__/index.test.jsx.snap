// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> should match the snapshot 1`] = `
<span
  aria-hidden="true"
  className="spinner-border"
  role="status"
  style={Object {}}
/>
`;
